### PR TITLE
fix(amplify-category-auth): add lambda with AdminQueries API permissions

### LINKED ISSUE #6576: AdminQueries Output 'ApiId' not found in stack
> **Note**: If your issue/bug is regarding the AWS Amplify Console service, please log it in the
> [Amplify Console GitHub Issue Tracker](https://github.com/aws-amplify/amplify-console/issues)

**Describe the bug**

I am trying to update a function with permissions to the `AdminQueries` api. The operation fails because the  `AdminQueries` api does not output it's Id.

The error:
```
UPDATE_FAILED   <function name> AWS::CloudFormation::Stack <time> Output 'ApiId' not found in stack 'arn:aws:cloudformation:<regionName>:<accountId>:stack/<projectId>-apiAdminQueries-<uuid>'
```

`Outputs` from `admin-queries-cloudformation-template.json`
```JSON
    "Outputs": {
        "RootUrl": {
            "Description": "Root URL of the API gateway",
            "Value": {"Fn::Join": ["", ["https://", {"Ref": "AdminQueries"}, ".execute-api.", {"Ref": "AWS::Region"}, ".amazonaws.com/", {"Fn::If": ["ShouldNotCreateEnvResources","Prod", {"Ref": "env"} ]}]]}
        },
        "ApiName": {
          "Description": "API Friendly name",
          "Value": "AdminQueries"
        }
    }
```

**Amplify CLI Version**
You can use `amplify -v` to check the amplify cli version on your system
4.41.2

**To Reproduce**
Steps to reproduce the behavior or terminal output if applicable
```
$ amplify update function
? Select which capability you want to update: Lambda function (serverless function)
? Select the Lambda function you want to update <function name>
? Which setting do you want to update? Resource access permissions
? Select the category api
? Api has # resources in this project. Select the one you would like your Lambda to access AdminQueries
? Select the operations you want to permit for AdminQueries create, read, update, delete
You can access the following resource attributes as environment variables from your Lambda function
        API_ADMINQUERIES_APIID
        API_ADMINQUERIES_APINAME
```

`amplify push` then fails. 

**Expected behavior**

I expect the cloud formation step to reach `UPDATE_COMPLETE`

**Screenshots**

**Desktop (please complete the following information):**

- OS: [e.g. Mac/Windows/Ubuntu]
  - WSL2 
- Node Version. You can use `node -v` to check the node version on your system
  - v14.15.3

**Log output**
Include any relevant log output under `~/.amplify/logs/amplify-cli-<issue-date>.log`

**Additional context**

### PR DESCRIPTION
**Pull request by sharmaaditya570191**
#### Description of changes

Fixes bug in creating a function with permissions to the AdminQueries API. The operation fails because the AdminQueries API does not output it's Id.

#### Issue #, if available

Fixes #6576

#### Description of how you validated changes

Added integration test to verify if lambda is added with AdminQueries API permissions.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by ammarkarachi** [comment]
cc: @akshbhu @cjihrig

File: packages/amplify-category-auth/resources/adminAuth/admin-queries-api-template.json.ejs [hunk 1]
```diff
@@ -290,6 +290,10 @@
         "ApiName": {
           "Description": "API Friendly name",
           "Value": "AdminQueries"
+        },
+        "ApiId": {
+          "Description": "API Id",
+          "Value": "AdminQueriesApiId"
```
Comment (cjihrig) [inline packages/amplify-category-auth/resources/adminAuth/admin-queries-api-template.json.ejs:296]: Should this be the `AdminQueries`'s API ID?

File: packages/amplify-category-auth/resources/adminAuth/admin-queries-api-template.json.ejs [hunk 2]
```diff
@@ -290,6 +290,10 @@
         "ApiName": {
           "Description": "API Friendly name",
           "Value": "AdminQueries"
+        },
+        "ApiId": {
+          "Description": "API Id",
```
Comment (cjihrig) [inline packages/amplify-category-auth/resources/adminAuth/admin-queries-api-template.json.ejs:295]:
```suggestion
          "Description": "API ID",
```

Comment (sharmaaditya570191) [reply packages/amplify-category-auth/resources/adminAuth/admin-queries-api-template.json.ejs:295]: Sure. I'll just update that.

**Review by kaustavghosh06 — APPROVED** [review]
Small nit as @cjihrig  mentioned - Change description to - "AdminQueries's API ID"
Great work :D

**Review by kaustavghosh06 — CHANGES_REQUESTED** [review]
Output should be {"Ref": "AdminQueries"} - discussed offline with @cjihrig who brought this up.

File: packages/amplify-category-auth/resources/adminAuth/admin-queries-api-template.json.ejs (hunk 1 above)
Comment (cjihrig) [reply packages/amplify-category-auth/resources/adminAuth/admin-queries-api-template.json.ejs:296]: Can you please update this to `{"Ref": "AdminQueries"}`. I chatted with @kaustavghosh06, and I think that's the value we want here.

Comment (sharmaaditya570191) [reply packages/amplify-category-auth/resources/adminAuth/admin-queries-api-template.json.ejs:296]: Sure. You mean replace line 296 with `{"Ref": "AdminQueries"}` right?

Comment (cjihrig) [reply packages/amplify-category-auth/resources/adminAuth/admin-queries-api-template.json.ejs:296]: Yes, that entire line would be:

```
"Value": {"Ref": "AdminQueries"}
```

**Review by cjihrig — APPROVED** [review]

**Review by kaustavghosh06 — APPROVED** [review]

**Pull request merged**